enclave-runtime/src/test/mocks/trusted_operation_pool_mock.rs

```diff
@@ -0,0 +1,229 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+	Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{
+	rpc::error,
+	top_pool::{
+		base_pool::TrustedOperation,
+		error::Error,
+		primitives::{
+			ImportNotificationStream, PoolFuture, PoolStatus, TrustedOperationPool,
+			TrustedOperationSource, TxHash,
+		},
+	},
+};
+use codec::Encode;
+use core::{future::Future, pin::Pin};
+use frame_support::sp_runtime::{
+	generic::BlockId,
+	traits::{BlakeTwo256, Hash, NumberFor},
+};
+use ita_stf::{ShardIdentifier, TrustedOperation as StfTrustedOperation};
+use itp_types::{Block, BlockHash as SidechainBlockHash, H256};
+use jsonrpc_core::futures::future::ready;
+use std::{
+	boxed::Box,
+	collections::HashMap,
+	string::String,
+	sync::{Arc, SgxRwLock as RwLock},
+	vec::Vec,
+};
+
+/// Mock for the trusted operation pool
+///
+/// To be used in unit tests
+pub struct TrustedOperationPoolMock {
+	submitted_transactions: RwLock<HashMap<ShardIdentifier, TxPayload>>,
+}
```

Comment:
Mock for the trusted operation pool. This was quite a bit of work, since I needed some basic functionality here in order for the clients of the top pool to work in a testing environment.